### PR TITLE
[RLC] add-custom-base-url-test-for-llc

### PR DESCRIPTION
**Pull request by qiaozha**
This PR is to add functionality test customBaseUrl from test server for RLC.

### CODE REVIEW

File: test/integration/customUrlRest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import CustomUrlRestClient, { CustomUrlRestClientRestClient } from "./generated/customUrlRest/src";
+import { assert } from 'chai';
+
+describe("Custom Endpoint", () => {
+  let client: CustomUrlRestClientRestClient;
+  let clientOptions: any;
+  beforeEach(() => {
+    clientOptions = { host: "host:3000", allowInsecureConnection: true };
```
Comment (sarangan12) [inline test/integration/customUrlRest.spec.ts:8]: From what I see in the client options, there is no `host` property. We have only `baseUrl`. How are you setting this property? I think this  might be the reason the CI is failing with the error message:

```
TypeError [ERR_INVALID_URL]: Invalid URL: http://{accountName}[object Object]//customuri
```

Could you please check and fix it? Does this code run locally?

**Review by sarangan12 — CHANGES_REQUESTED** [review]
nit: Please add a description on the issue detailing what and why the changes are done for this PR. Thanks

File: test/integration/customUrlRest.spec.ts (hunk 1 above)
Comment (qiaozha) [reply test/integration/customUrlRest.spec.ts:8]: yeah, I am still trying to understand why

File: test/integration/customUrlRest.spec.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+describe("CustomRest Endpoint", () => {
+  let client: CustomUrlRestClientRestClient;
+  let clientOptions: any;
+  beforeEach(() => {
+    clientOptions = { allowInsecureConnection: true };
+    client = CustomUrlRestClient("host:3000", clientOptions);
+  });
+
+  it("should return 200", async () => {
+    const result = await client.path("/customuri", 'local').get();
```
Comment (joheredi) [inline test/integration/customUrlRest.spec.ts:13]: According to the conversation in https://github.com/Azure/autorest.typescript/issues/1057

I'm expecting this call to look like this
```typescript
// Internally as defined in the swagger, the base url has parameters in this case the base url is: `{accountName}{suffix}`, 
// where suffix is a global parameter that we pass when building the client as above, and `accountName` is a method 
// parameter that we'll require as part of the operation options.
await client.path("/customuri").get({pathParameters: { accountName: "local"} })
```

For this we need to:

1. Add a new `pathParameters` property to `RequestParameters` in @azure-rest/core-client`
2. Update logic in `@azure-rest/core-client` to use `pathParameters` when building the url.
3. Generate the parameter types to include `accountName` as a pathParameter

For the changes in (2) you'd need to add logic before this loop https://github.com/Azure/azure-sdk-for-js/blob/a8b7f42993733e43a3dbaee89e66393df1b8bfa0/sdk/core/core-client-rest/src/urlHelpers.ts#L27

In which you loop the object options.pathParameters (added as part of (1)) to replace any occurrences of its keys in the URL with the value.

Comment (qiaozha) [reply test/integration/customUrlRest.spec.ts:13]: Got it. Thanks for info and instructions

**Comment by qiaozha** [comment]
@joheredi @sarangan12 Could you help review this PR again, I have test it with https://github.com/Azure/azure-sdk-for-js/pull/19463 in my local and it works. Thanks

File: test/integration/customUrlRest.spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+describe("CustomRest Endpoint", () => {
+  let client: CustomUrlRestClientRestClient;
+  let clientOptions: any;
+  beforeEach(() => {
+    clientOptions = { allowInsecureConnection: true };
+    client = CustomUrlRestClient("host:3000", clientOptions);
+  });
+
+  it("should return 200", async () => {
```
Comment (joheredi) [inline test/integration/customUrlRest.spec.ts:12]: Can you also add a test using the shortcut methods?

File: test/utils/test-swagger-gen.ts [hunk 4]
```diff
@@ -265,6 +265,17 @@ const testSwaggers: { [name: string]: SwaggerConfig } = {
     addCredentials: false,
     isTestPackage: true
   },
+  customUrlRest: {
```
Comment (joheredi) [inline test/utils/test-swagger-gen.ts:268]: Would you mind moving this to the bottom next to all the other REST Client tests?

**Review by joheredi — APPROVED** [review]
Looks good, left a couple of minor comments

File: test/integration/customUrlRest.spec.ts (hunk 3 above)
Comment (qiaozha) [reply test/integration/customUrlRest.spec.ts:12]: added

File: test/utils/test-swagger-gen.ts (hunk 4 above)
Comment (qiaozha) [reply test/utils/test-swagger-gen.ts]: moved

**Comment by qiaozha** [comment]
@sarangan12 @joheredi Could you take a look at this PR again ? Thanks

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**